packages/@aws-cdk/aws-wafv2/lib/ipset.ts

```diff
@@ -0,0 +1,111 @@
+import * as cdk from '@aws-cdk/core';
+import * as waf from './index';
+
+export enum IPAddressVersion {
+    IPV4 = 'IPV4',
+    IPV6 = 'IPV6'
+}
+
+export enum IPSetScope {
+    CLOUDFRONT = 'CLOUDFRONT',
+    REGIONAL = 'REGIONAL'
+}
+
+export interface IIPSet {
+    /**
+     * A friendly name of the IP set. You cannot change the name of an IPSet
+     * after you create it.
+     */
+    readonly ipSetName: string;
+}
+
+export abstract class IPSetBase extends cdk.Resource implements IIPSet {
+    /**
+     * A friendly name of the IP set. You cannot change the name of an IPSet
+     * after you create it.
+     */
+    public abstract readonly ipSetName: string;
+}
+
+export interface IPSetProps {
+    /**
+     * A friendly name of the IP set. You cannot change the name of an IPSet
+     * after you create it.
+     */
+    readonly name?: string;
+
+    /**
+     * A friendly description of the IP set. You cannot change the description
+     * of an IP set after you create it.
+     */
+    readonly description?: string;
+
+    /**
+     * Contains an array of strings that specify one or more IP addresses or
+     * blocks of IP addresses in Classless Inter-Domain Routing (CIDR) notation.
+     * AWS WAF supports all address ranges for IP versions IPv4 and IPv6.
+     */
+    readonly addresses?: string[];
+
+    /**
+     * Specify IPV4 or IPV6.
+     *
+     * @efault IPV4
+     */
+    readonly ipAddressVersion: IPAddressVersion;
+
+    /**
+     * Specifies whether this is for an AWS CloudFront distribution or for a
+     * regional application. A regional application can be an Application
+     * Load Balancer (ALB) or an API Gateway stage.
+     *
+     * @default CLOUDFRONT
+     */
+    readonly scope: IPSetScope;
+}
+
+export class IPSet extends IPSetBase {
+    /**
+     * A friendly name of the IP set. You cannot change the name of an IPSet
+     * after you create it.
+     */
+    public readonly ipSetName: string;
+
+    /**
+     * Contains an array of strings that specify one or more IP addresses or
+     * blocks of IP addresses in Classless Inter-Domain Routing (CIDR) notation.
+     * AWS WAF supports all address ranges for IP versions IPv4 and IPv6.
+     */
+    private addresses: string[] = [];
+
+    constructor(scope: cdk.Construct, id: string, props: IPSetProps) {
+        super(scope, id, {
+            physicalName: props.name || cdk.PhysicalName.GENERATE_IF_NEEDED
+        });
+
+        this.ipSetName = this.physicalName;
+
+        this.addAddresses(...props.addresses || []);
+
+        const resource: waf.CfnIPSet = new waf.CfnIPSet(this, 'IPSet', {
```

Comment:
Should be called `Resource` instead of `IPset` imho

Reply:
Sorry did you already start work on this then? I checked through the issues and didn't see things related to this.

I followed the lambda function for that line, i'm happy to change it. It's hard to tell what standard new things should follow.

Reply:
i've updated this to match what you wanted.

Reply:
Maybe @eladb could clarify this but I recall that nested L1s are called `Resource`. I did not yet start on ann implementation because I hit the Ìnternal Failure` on first check.